packages/bitcoin/src/fungible.ts

```diff
@@ -0,0 +1,75 @@
+import { BaseTransfer, Config } from '@buildwithsygma/core';
+import type { Config as TConfig, BitcoinResource, Domain } from '@buildwithsygma/core/types';
+import type { networks } from 'bitcoinjs-lib';
+
+import type {
+  BitcoinTransferParams,
+  BitcoinTransferRequest,
+  TypeOfAddress,
+  UTXOData,
+} from './types.js';
+import { getPsbt } from './utils/index.js';
+
+export async function createBitcoinFungibleTransfer(
+  params: BitcoinTransferParams,
+): Promise<BitcoinFungibleTransfer> {
+  const config = new Config();
+  await config.init(process.env.SYGMA_ENV || params.environment);
+  return new BitcoinFungibleTransfer(params, config);
+}
+
+class BitcoinFungibleTransfer extends BaseTransfer {
+  protected publicKey: Buffer;
+  protected typeOfAddress: TypeOfAddress;
+  protected network: networks.Network;
+  protected changeAddress?: string;
+  protected feeRate: bigint;
+  protected utxoData: UTXOData[];
+  protected size: bigint;
+  protected destinationAddress: string;
+  protected amount: bigint;
+  protected sourceDomain: Domain;
+  protected destinationDomain: Domain;
+  protected feeAddress: string;
+  protected feeAmount: bigint;
+
+  constructor(transfer: BitcoinTransferParams, config: TConfig) {
+    super(transfer, config);
+    this.destinationAddress = transfer.destinationAddress;
+    this.amount = transfer.amount;
+    this.publicKey = transfer.publicKey;
+    this.typeOfAddress = transfer.typeOfAddress;
+    this.network = transfer.network;
+    this.changeAddress = transfer.changeAddress;
+    this.feeRate = transfer.feeRate;
+    this.utxoData = transfer.utxoData;
+    this.size = transfer.size;
+    this.sourceDomain = config.getDomain(transfer.source);
+    this.destinationDomain = config.getDomain(transfer.destination);
+
+    this.feeAddress = this.sourceDomain.feeAddress as string;
+    this.feeAmount = BigInt((this.resource as BitcoinResource).feeAmount!);
```

Comment:
is `feeAmount` hardcoded?

Reply:
in the shared config? yes [here](https://github.com/sygmaprotocol/sygma-shared-configuration/blob/main/testnet/shared-config-test.json#L694)